### PR TITLE
build: show ninja stats for testing/debug builds

### PR DESCRIPTION
**Pull request by jkleinsc**
This PR adds ninja stats output for testing and debug builds.  This is primarily informational,  but could be useful in diagnosing where our [builds are slow](https://chromium.googlesource.com/chromium/src/+/master/docs/windows_build_instructions.md#Why-is-my-build-slow).  

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: no-notes

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -102,6 +102,9 @@ env-mac-large: &env-mac-large
 env-disable-crash-reporter-tests: &env-disable-crash-reporter-tests
   DISABLE_CRASH_REPORTER_TESTS: true
 
+env-ninja-status: &env-ninja-status
+  NINJA_STATUS: "[%r processes, %f/%t @ %o/s : %es ]"
```
Comment (nornagon) [inline .circleci/config.yml]:
```suggestion
  NINJA_STATUS: "[%r processes, %f/%t @ %o/s : %es]"
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -535,6 +538,13 @@ step-fix-known-hosts-linux: &step-fix-known-hosts-linux
         ./src/electron/.circleci/fix-known-hosts.sh
       fi
 
+step-ninja-summary: &step-ninja-summary
+  run:
+    name: Print ninja summary
+    command: |
+      cd src
+      python third_party/depot_tools/post_build_ninja_summary.py -C out/Default
```
Comment (nornagon) [inline .circleci/config.yml]: any particular reason to use the vendored depot_tools instead of the one that's installed on CI?

Comment (jkleinsc) [reply .circleci/config.yml]: I saw this comment and thought it was always being checked out to third_party: https://github.com/electron/electron/blob/f5339e15b13cab7b217c7c53f58259c333218333/.circleci/config.yml#L555

File: .circleci/config.yml [hunk 3]
```diff
@@ -102,6 +102,9 @@ env-mac-large: &env-mac-large
 env-disable-crash-reporter-tests: &env-disable-crash-reporter-tests
   DISABLE_CRASH_REPORTER_TESTS: true
 
+env-ninja-status: &env-ninja-status
+  NINJA_STATUS: "[%r processes, %f/%t @ %o/s : %es]"
```
Comment (nornagon) [inline .circleci/config.yml]: hm, i see now that this stuff comes from autoninja, which also has that extra space there? I wonder if we should just invoke `autoninja`. either way i was probably wrong about removing the space.
```suggestion
  NINJA_STATUS: "[%r processes, %f/%t @ %o/s : %es ]"
```

Comment (jkleinsc) [reply .circleci/config.yml]: It works either way 😁

File: appveyor.yml [hunk 4]
```diff
@@ -50,6 +50,7 @@ build_script:
       if(($env:APPVEYOR_PULL_REQUEST_HEAD_REPO_NAME -split "/")[0] -eq ($env:APPVEYOR_REPO_NAME -split "/")[0]) {
         Write-warning "Skipping PR build for branch"; Exit-AppveyorBuild
       }
+  - where gclient
```
Comment (MarshallOfSound) [inline appveyor.yml]: Is this needed or was it for debugging?

Comment (jkleinsc) [reply appveyor.yml]: debugging

**Comment by MarshallOfSound** [comment]
As part of this can we add `out/{OUT_DIR}/.ninja_log` to the artifacts list.  It's really helpful for understanding performance of the build as well

**Pull request merged**